qiime2/sdk/usage.py

```diff
@@ -238,6 +238,16 @@ def init_metadata(self, ref, factory):
     def _init_metadata_(self, ref, factory):
         raise NotImplementedError
 
+    def init_data_collection(self, ref, records):
```

Comment:
I don't think that this approach will work, since you are passing in the factories now, rather than `ScopeRecord`s. I think we need scope records, because those will have `ref`s associated with them, which will let us name the things in our various implementations. Without that info, I think you'll be left guessing _what_ to name them in the driver implementations.

Reply:
I will push up another iteration rather than discussing here.  I think that will be easier.